### PR TITLE
UIDATIMP-396: Remove extra action buttons in profile setting screens

### PR DESCRIPTION
**Pull request by IhorKoba**
## Purpose

- Update pane header after [PR](https://github.com/folio-org/stripes-components/pull/1112) since buttons with the same functions are duplicated

## Approach

- Add renderHeader prop to `<Pane>` component
- Move props from  `<Pane>` to `<PaneHeader>` component
- Remove lastMenu prop from `<Pane>` component

## Issues
[UIDATIMP-394](https://issues.folio.org/browse/UIDATIMP-394)
[UIDATIMP-395](https://issues.folio.org/browse/UIDATIMP-395)
[UIDATIMP-396](https://issues.folio.org/browse/UIDATIMP-396)
[UIDATIMP-397](https://issues.folio.org/browse/UIDATIMP-397)

### CODE REVIEW

File: src/components/SearchAndSort/SearchAndSort.js [hunk 1]
```diff
@@ -753,19 +718,25 @@ export class SearchAndSort extends Component {
       />
     );
 
+    const header = renderProps => (
```
Comment (mariia-aloshyna) [inline src/components/SearchAndSort/SearchAndSort.js]: Move this function to the class methods and rename it to "renderPaneHeader" for example, please.

File: src/components/Spinner/Spinner.js [hunk 2]
```diff
@@ -1,30 +1,40 @@
...
-  >
-    <Preloader />
-  </Pane>
-));
+}) => {
+  const header = renderProps => (
+    <PaneHeader
+      {...renderProps}
+      id={paneId}
+      paneTitle=""
```
Comment (mariia-aloshyna) [inline src/components/Spinner/Spinner.js]: This prop can be omitted.

File: src/settings/ActionProfiles/ViewActionProfile.js [hunk 3]
```diff
@@ -175,6 +157,18 @@ export class ViewActionProfile extends Component {
       </AppIcon>
     );
 
+    const header = renderProps => (
```
Comment (mariia-aloshyna) [inline src/settings/ActionProfiles/ViewActionProfile.js]: Move this function to the class methods and rename it to "renderPaneHeader" for example, please.

File: src/settings/JobProfiles/ViewJobProfile.js [hunk 4]
```diff
@@ -148,21 +141,12 @@ export class ViewJobProfile extends Component {
...
-      PropTypes.string.isRequired,
-    ]),
     tagsEnabled: PropTypes.bool,
     onClose: PropTypes.func.isRequired,
     onDelete: PropTypes.func.isRequired,
     paneId: PropTypes.string, // eslint-disable-line
     ENTITY_KEY: PropTypes.string, // eslint-disable-line
     actionMenuItems: PropTypes.arrayOf(PropTypes.string), // eslint-disable-line
-    withEditRecordButton: PropTypes.bool,
-    withRunRecordButton: PropTypes.bool,
```
Comment (mariia-aloshyna) [inline src/settings/JobProfiles/ViewJobProfile.js]: It seems like these props are unused. Remove these props from the `UploadingJobsDisplay.js` file, please.

File: src/settings/JobProfiles/ViewJobProfile.js [hunk 5]
```diff
@@ -226,10 +208,6 @@ export class ViewJobProfile extends Component {
     this.setState({ showDeleteConfirmation: true });
   };
 
-  showRunConfirmation = () => {
```
Comment (mariia-aloshyna) [inline src/settings/JobProfiles/ViewJobProfile.js]: This method is used by the **Run** button, which is one of the action buttons (see it in `menuTemplate.js` file 71 line), do not remove it, please.

File: src/settings/JobProfiles/ViewJobProfile.js [hunk 6]
```diff
@@ -402,17 +341,23 @@ export class ViewJobProfile extends Component {
       ? this.renderActionMenu
       : null;
 
-    return (
-      <Pane
+    const header = renderProps => (
```
Comment (mariia-aloshyna) [inline src/settings/JobProfiles/ViewJobProfile.js]: Move this function to the class methods and rename it, please.

File: src/settings/MappingProfiles/ViewMappingProfile.js [hunk 7]
```diff
@@ -190,17 +171,23 @@ export class ViewMappingProfile extends Component {
     const tagsEntityLink = `data-import-profiles/mappingProfiles/${mappingProfile.id}`;
     const associations = [...[], ...mappingProfile.parentProfiles, ...mappingProfile.childProfiles];
 
-    return (
-      <Pane
+    const header = renderProps => (
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/ViewMappingProfile.js]: Move this function to the class methods and rename it, please

File: src/settings/MatchProfiles/ViewMatchProfile.js [hunk 8]
```diff
@@ -298,17 +278,23 @@ export class ViewMatchProfile extends Component {
       },
     };
 
-    return (
-      <Pane
+    const header = renderProps => (
```
Comment (mariia-aloshyna) [inline src/settings/MatchProfiles/ViewMatchProfile.js]: Move this function to the class methods and rename it, please

File: src/components/SearchAndSort/SearchAndSort.js (hunk 1 above)
Comment (IhorKoba) [reply src/components/SearchAndSort/SearchAndSort.js]: Done!

File: src/components/Spinner/Spinner.js (hunk 2 above)
Comment (IhorKoba) [reply src/components/Spinner/Spinner.js]: Done!

File: src/settings/ActionProfiles/ViewActionProfile.js (hunk 3 above)
Comment (IhorKoba) [reply src/settings/ActionProfiles/ViewActionProfile.js]: Done!

File: src/settings/JobProfiles/ViewJobProfile.js (hunk 4 above)
Comment (IhorKoba) [reply src/settings/JobProfiles/ViewJobProfile.js]: Done!

File: src/settings/JobProfiles/ViewJobProfile.js (hunk 5 above)
Comment (IhorKoba) [reply src/settings/JobProfiles/ViewJobProfile.js]: Done!

File: src/settings/JobProfiles/ViewJobProfile.js (hunk 6 above)
Comment (IhorKoba) [reply src/settings/JobProfiles/ViewJobProfile.js]: Done!

File: src/settings/MappingProfiles/ViewMappingProfile.js (hunk 7 above)
Comment (IhorKoba) [reply src/settings/MappingProfiles/ViewMappingProfile.js]: Done!

File: src/settings/MatchProfiles/ViewMatchProfile.js (hunk 8 above)
Comment (IhorKoba) [reply src/settings/MatchProfiles/ViewMatchProfile.js]: Done!

File: src/settings/ActionProfiles/ViewActionProfile.js [hunk 9]
```diff
@@ -175,6 +148,32 @@ export class ViewActionProfile extends Component {
...
+        actionMenu={this.renderActionMenu}
+        dismissible
+        onClose={onClose}
+      />
+    );
+  };
+
+  render() {
+    const { tagsEnabled } = this.props;
+
```
Comment (VictorSoroka1) [inline src/settings/ActionProfiles/ViewActionProfile.js]: Please remove a line here and add after the state [here](https://github.com/folio-org/ui-data-import/pull/434/files#diff-63e6fc715be2fbb76352e74f6e40fc4cR341). Usually variables destructured from state and props are grouped and then they are followed with a black line.

Comment (IhorKoba) [reply src/settings/ActionProfiles/ViewActionProfile.js]: Done!

File: src/settings/JobProfiles/ViewJobProfile.js [hunk 10]
```diff
@@ -307,61 +289,56 @@ export class ViewJobProfile extends Component {
...
-      </PaneMenu>
-    );
-  }
-
-  renderLastMenu(record) {
+  renderPaneHeader = renderProps => {
     const {
-      withEditRecordButton,
-      withRunRecordButton,
+      onClose, actionMenuItems,
```
[the remark this answers is not in the document]
Comment (IhorKoba) [reply src/settings/JobProfiles/ViewJobProfile.js]: Done!

**Pull request merged**